indexer/src/main/java/consumer/IndexerConsumer.java

```
@@ -45,34 +51,41 @@ public void consume() {
-                            .source(rValAsMap);
-                    request.add(indexRequest);
-
-                    logger.debug("Record was indexed: (%s, %s, %d, %d)\n",
-                            r.key(), r.value(), r.partition(), r.offset());
-                } catch (Exception e) {
-                    e.printStackTrace();
-                }
-            });
+
+            BulkRequest request = getBulkRequestForRecords(records);
 
             if (request.numberOfActions() == 0) continue;
             try {
                 BulkResponse bulkResponse =elasticsearchClient.bulk(request, RequestOptions.DEFAULT);
                 if (bulkResponse.hasFailures()) {
 
-                    logger.debug("Bulk: an error has occured in bulk response");
+                    logger.debug("Bulk: an error has occured in bulk response " + bulkResponse.buildFailureMessage());
                 }
                 consumer.commitAsync();
-            } catch (IOException e) {
+            } catch (Exception e) {
                 e.printStackTrace();
             }
         }
     }
+
+    private BulkRequest getBulkRequestForRecords(ConsumerRecords<String, String> records) {
+        BulkRequest bulkRequest = new BulkRequest();
+        records.forEach(r -> {
+
+            // index record to ES
+            try {
+                Map<String, String> rValAsMap = JsonParser.getObjFromJsonString(r.value(), Map.class);
+
+                Account account = accountsServiceApi.getAccount(r.key());
+                IndexRequest indexRequest = new IndexRequest(account.getEsindex(), "_doc")
+                        .source(rValAsMap);
+                bulkRequest.add(indexRequest);
+
+                logger.debug("Record was added to bulk request: ({}, {}, {}, {})\n", r.key(), r.value(), r.partition(), r.offset());
+            } catch (Exception | AccountInvalidTokenException e) {
+                e.printStackTrace();
```

Looks like you swallow an exception here - I guess you want to return to the client some kind of error response

---

which kind of error response? throwing exceptions invoked in the foreach loop is apparently forbidden.